### PR TITLE
Implement advanced login

### PR DESCRIPTION
**Pull request by thanhconguet214**

### CODE REVIEW

**Comment by thanhconguet214** [comment]
ready

File: app/assets/stylesheets/custom.scss [hunk 1]
```diff
@@ -174,4 +174,18 @@ input {
   }
 }
 
+.checkbox {
+  margin-top: -10px;
+  margin-bottom: 10px;
```
Comment (mdmahmudurrahman) [inline app/assets/stylesheets/custom.scss]: You can combine line 179 and 178 into 1 line

File: app/helpers/sessions_helper.rb [hunk 2]
```diff
@@ -1,17 +1,43 @@
 module SessionsHelper
+<<<<<<< HEAD
   def log_in user 
+=======
+  def log_in user
+>>>>>>> Implement advanced login
```
Comment (mdmahmudurrahman) [inline app/helpers/sessions_helper.rb]: Remove those strings

File: app/helpers/sessions_helper.rb [hunk 3]
```diff
@@ -1,17 +1,43 @@
 module SessionsHelper
+<<<<<<< HEAD
   def log_in user 
+=======
```
Comment (mdmahmudurrahman) [inline app/helpers/sessions_helper.rb]: Remove

File: app/helpers/sessions_helper.rb [hunk 4]
```diff
@@ -1,17 +1,43 @@
 module SessionsHelper
+<<<<<<< HEAD
```
Comment (mdmahmudurrahman) [inline app/helpers/sessions_helper.rb]: Remove and be careful while rebasing.

File: app/models/user.rb [hunk 5]
```diff
@@ -14,5 +15,23 @@ def digest string
...
+    update_attributes remember_digest: User.digest(remember_token)
+  end
+
+  def authenticated? remember_token
+    return false if remember_digest.nil?
+    BCrypt::Password.new(remember_digest).is_password? remember_token
+  end
+
+  def forget
+    update_attribute remember_digest: nil
```
Comment (mdmahmudurrahman) [inline app/models/user.rb]: 1. Please use update_attributes
2. Can you tell me all of the difference between update_attribute and update_attributes here.

File: app/views/sessions/new.html.erb [hunk 6]
```diff
@@ -7,6 +7,10 @@
       <%= f.email_field :email, class: "form-control" %>
       <%= f.label :password %>
       <%= f.password_field :password, class: "form-control" %>
+      <%= f.label :remember_me, class: "checkbox inline" do %>
+        <%= f.check_box :remember_me %>
+        <span><%= t(".Remember") %></span>
```
Comment (mdmahmudurrahman) [inline app/views/sessions/new.html.erb]: Please optional remove ()

**Comment by mdmahmudurrahman** [comment]
commented

File: app/models/user.rb (hunk 5 above)
Comment (thanhconguet214) [reply app/models/user.rb]: "update_attribute" update one atribute and it get 2 parameters 
"update_attributes" update desired attributes  and it get hash table

**Comment by thanhconguet214** [comment]
ready

File: app/helpers/sessions_helper.rb [hunk 7]
```diff
@@ -3,15 +3,37 @@ def log_in user
...
+    cookies.permanent.signed[:user_id] = user.id
+    cookies.permanent[:remember_token] = user.remember_token
+  end
+
   def current_user
-    @current_user ||= User.find_by id: session[:user_id]
+    if user_id = session[:user_id]
+      @current_user ||= User.find_by id: user_id
+    elsif user_id = cookies.signed[:user_id]
+      raise
```
Comment (huuphuc93) [inline app/helpers/sessions_helper.rb]: what raise? remove if you don't use

**Comment by huuphuc93** [comment]
commented
comment nhỏ

**Comment by thanhconguet214** [comment]
ready

**Comment by huuphuc93** [comment]
merged
this (pr) can be (merged)  now
CC: [To:2076564] Nguyen Huy Hung C (93)
CC: [To:1546746] Md. Mahmudur Rahman (Mahmud)
CC: [To:1818111] Nguyen Thi Huong (94)

**Pull request merged**